### ISSUE TITLE
Error in opening Newick-format file

### ISSUE THREAD

**Issue by vsundaram** [biopython/biopython#810]
Hi, 

I get the following error while trying to open a newick-format file: Bio.Phylo.NewickIO.NewickError: Number of open/close parentheses do not match.

I've worked on this file in the past (late March), and it worked just fine. The Bio module was updated recently (to version 1.66), and I thought the change in versions was the issue. But, when I installed and ran version 1.65, I still get the same error. I don't think the version has change from late March till now. 

This might be a trivial problem. I'd appreciate any help with this.

Thank you.
Vasavi

**Comment by peterjc** [comment]
Can you count the number of open and closing brackets (parentheses), using a text editor or Python? M guess is the file has been edited and a typo was added somehow? Or can you share the file with us?

**Comment by vsundaram** [comment]
Thanks ! I didn't mention this earlier - I have not edited the file. And, I
tried it on various other files to rule-out the issue of that file being
problematic.

I just checked the file(s) and they have the same number of brackets.

On Fri, Apr 15, 2016 at 6:31 PM, Peter Cock <notifications@github.com>
wrote:

> Can you count the number of open and closing brackets (parentheses), using
> a text editor or Python? M guess is the file has been edited and a typo was
> added somehow? Or can you share the file with us?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/biopython/biopython/issues/810#issuecomment-210683188>
>



-- 
Vasavi Sundaram
PhD Candidate - Computational and Systems Biology
Ting Wang Lab | Center for Genome Sciences | Dept. of Genetics
Washington University in St. Louis

**Comment by vsundaram** [comment]
After trying a few variations, I've solved the problem - there were
semi-colon (";") in my sequence identifier that were causing the error.
After I removed all of the semi-colons in the file, it works now. The error
message (Bio.Phylo.NewickIO.NewickError: Number of open/close parentheses
do not match.) confused me.

However it confuses me that the last time I ran Phylo.read() (several weeks
ago), I didn't have this problem and there were/are semi-colons in the
file. Has there been a change in the code?

Thanks !

On Fri, Apr 15, 2016 at 6:36 PM, Vasavi Sundaram <vasavi.sundaram@gmail.com>
wrote:

> Thanks ! I didn't mention this earlier - I have not edited the file. And,
> I tried it on various other files to rule-out the issue of that file being
> problematic.
>
> I just checked the file(s) and they have the same number of brackets.
>
> On Fri, Apr 15, 2016 at 6:31 PM, Peter Cock <notifications@github.com>
> wrote:
>
>> Can you count the number of open and closing brackets (parentheses),
>> using a text editor or Python? M guess is the file has been edited and a
>> typo was added somehow? Or can you share the file with us?
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/biopython/biopython/issues/810#issuecomment-210683188>
>>
>
>
>
> --
> Vasavi Sundaram
> PhD Candidate - Computational and Systems Biology
> Ting Wang Lab | Center for Genome Sciences | Dept. of Genetics
> Washington University in St. Louis
>



-- 
Vasavi Sundaram
PhD Candidate - Computational and Systems Biology
Ting Wang Lab | Center for Genome Sciences | Dept. of Genetics
Washington University in St. Louis

**Comment by peterjc** [comment]
Hi @vsundaram - special punctuation can often cause problems. Do you know what the old version of Biopython you were using was? I looked back over the code history in GitHub and didn't spot anything relevant here.

**Comment by vsundaram** [comment]
Hi, I believe I was using the latest version of BioPhython in March. We
updated again in April, I believe. I'm not sure, and will let you know if I
can find out any more specific details.

On Sat, Apr 23, 2016 at 5:43 AM, Peter Cock <notifications@github.com>
wrote:

> Hi @vsundaram <https://github.com/vsundaram> - special punctuation can
> often cause problems. Do you know what the old version of Biopython you
> were using was? I looked back over the code history in GitHub and didn't
> spot anything relevant here.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/biopython/biopython/issues/810#issuecomment-213717467>
>



-- 
Vasavi Sundaram
PhD Candidate - Computational and Systems Biology
Ting Wang Lab | Center for Genome Sciences | Dept. of Genetics
Washington University in St. Louis

**Comment by cymon** [comment]
On 18 April 2016 at 15:50, Vasavi Sundaram <notifications@github.com> wrote:

> After trying a few variations, I've solved the problem - there were
> semi-colon (";") in my sequence identifier that were causing the error.
> After I removed all of the semi-colons in the file, it works now.


I'm pretty sure correct Newick tree format should be ended with a
semi-colon (although some software mistakenly fail to add it).

((a,b),(c,d));


C.

<http://orcid.org/0000-0002-4927-979X>

**Comment by peterjc** [comment]
Yes, the only semi-colons on http://evolution.genetics.washington.edu/phylip/newicktree.html are at the end. It is unclear from that document alone if they would ever be allowed in the names (and if so, if they need escaping).

**Comment by rajneeshdadwal** [comment]
> Hi @vsundaram - special punctuation can often cause problems. Do you know what the old version of Biopython you were using was? I looked back over the code history in GitHub and didn't spot anything relevant here.

the biopython version 1.78 causes a problem as to use some other program when I installed 1.78 instead of 1.79 there was similar problem

**Comment by peterjc** [comment]
@rajneeshdadwal you could retest with the current release, Biopython 1.81, but if I recall correctly nothing is likely to have changed in this part of the code.
